tests/Aspire.Hosting.Containers.Tests/FromDockerfileTests.cs

```diff
@@ -0,0 +1,358 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Aspire.Hosting.Dcp.Model;
+using Aspire.Hosting.Testing;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Aspire.Hosting.Containers.Tests;
+
+public class WithDockerfileTests
+{
+    [Fact]
+    public async Task WithDockerfileLaunchesContainerSuccessfully()
+    {
+        if (!IsDockerAvailable())
+        {
+            return;
```

Comment:
If you don't want to use `[RequiresDocker]` then I would suggest making this `[ConditionalFact]` and `throw new SkipException("..")`, so the test correctly shows up as `Skipped`.

Reply:
It doesn't seem to work that way. It just results in the test failing on Windows. Are we missing some infrastructure to make this work?

Reply:
I can take a look later this week.